.github/workflows/golang.yml

```
@@ -38,14 +38,33 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        FABRIC_VERSION:
-          - '2.5.3'
-        CREATE_CHANNEL:
-          - create_channel
-          - existing_channel
+        include:
+          - FABRIC_VERSION: '2.5.3'
+            CREATE_CHANNEL: 'create_channel'
+            CONSENSUS: RAFT
+          - FABRIC_VERSION: '2.5.3'
+            CREATE_CHANNEL: 'existing_channel'
+            CONSENSUS: RAFT
+          - FABRIC_VERSION: '3.0.0-preview'
+            CREATE_CHANNEL: 'create_channel'
+            CONSENSUS: RAFT
+          - FABRIC_VERSION: '3.0.0-preview'
+            CREATE_CHANNEL: 'existing_channel'
+            CONSENSUS: RAFT
+          - FABRIC_VERSION: '3.0.0-preview'
+            CREATE_CHANNEL: 'create_channel'
+            CONSENSUS: BFT
+          - FABRIC_VERSION: '3.0.0-preview'
+            CREATE_CHANNEL: 'existing_channel'
+            CONSENSUS: BFT
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
         with:
           go-version: '1.20'
-      - run: ${{ github.workspace }}/.github/scripts/ci.sh ${{matrix.FABRIC_VERSION}} ${{matrix.CREATE_CHANNEL}}
+      - name: run end to end test
+        run: |
+         export FABRIC_VERSION=${{matrix.FABRIC_VERSION}}
+         export CREATE_CHANNEL=${{matrix.CREATE_CHANNEL}}
+         export CONSENSUS=${{matrix.CONSENSUS}}
+         ${{ github.workspace }}/.github/scripts/ci.sh
```

This could be written using GitHub Actions directives rather than in-line shell scripting.
```suggestion
       env:
         FABRIC_VERSION: ${{matrix.FABRIC_VERSION}}
         CREATE_CHANNEL: ${{matrix.CREATE_CHANNEL}}
         CONSENSUS: ${{matrix.CONSENSUS}}
       run: ${{ github.workspace }}/.github/scripts/ci.sh
```